tests/sentry/api/test_organization_missing_members.py

```diff
@@ -86,9 +99,46 @@ def test_filters_old_commits(self):
         ]
 
     def test_no_authors(self):
-        org = self.create_organization()
+        org = self.create_organization(owner=self.create_user())
         self.create_member(user=self.user, organization=org, role="manager")
 
         response = self.get_success_response(org.slug)
         assert response.data[0]["integration"] == "github"
         assert response.data[0]["users"] == []
+
+    def test_not_common_domain(self):
+        user = self.create_user(email="owner@exampletwo.com")
+        self.create_member(
+            organization=self.organization,
+            user=user,
+            role="owner",
+        )
+        OrganizationMember.objects.filter(
+            organization_id=self.organization.id, role="owner", user_id=user.id
+        ).update(user_email="owner@exampletwo.com")
+
+        response = self.get_success_response(self.organization.slug)
+
+        assert response.data[0]["integration"] == "github"
+        assert response.data[0]["users"] == [
+            {"email": "c@example.com", "externalId": "c", "commitCount": 2},
+            {"email": "d@example.com", "externalId": "d", "commitCount": 1},
+            {"email": "a@exampletwo.com", "externalId": "not", "commitCount": 1},
+        ]
+
+    def test_query(self):
```

Comment:
Can we rename this test for clarity, I'm a little confused what it does.
Also, why does  `"d@example.com"` not show up in the response body here?

Reply:
i'm querying for `c@example.com`. one of the users in the body has `c@example.com` for their email, and the other one has `c@example.com` for their username. the user with email `d@example.com` doesn't match the query in either the username or email fields

Reply:
Probably `test_query_on_author_email_and_external_id` would be good. I would suggest that you add a comment that `self.nonmember_commit_author1` has the email already.